src/main/scala/zio/prelude/NonEmptyList.scala

```diff
@@ -174,17 +163,6 @@ sealed trait NonEmptyList[+A] { self =>
   final def foreach[F[+_]: AssociativeBoth: Covariant, B](f: A => F[B]): F[NonEmptyList[B]] =
     reduceMapRight(f(_).map(single))((a, fas) => f(a).zipWith(fas)(cons))
 
-  /**
-   * Returns the hashCode of this `NonEmptyList`.
-   */
-  override final def hashCode: Int = {
```

Comment:
Not sure about the performance implications of repeatedly calling the `hashCode` definition for product types here instead of using the one for collection types. Would be inclined to keep these.